diffrax/_solver/ros3p.py

```diff
@@ -0,0 +1,223 @@
+
+    def init(self, terms, t0, t1, y0, args) -> _SolverState:
+        del t1
+        return terms.vf(t0, y0, args)
+
+    def order(self, terms):
+        return 3
+
+    def step(
+        self,
+        terms: AbstractTerm[ArrayLike, ArrayLike],
+        t0: RealScalarLike,
+        t1: RealScalarLike,
+        y0: Y,
+        args: Args,
+        solver_state: _SolverState,
+        made_jump: BoolScalarLike,
+    ) -> tuple[Y, Y, DenseInfo, _SolverState, RESULTS]:
+        y0_leaves = jtu.tree_leaves(y0)
+        sol_dtype = jnp.result_type(*y0_leaves)
+
+        time_derivative = jax.jacfwd(lambda t: terms.vf(t, y0, args))(t0)
+        control = terms.contr(t0, t1)
+
+        γ = jnp.array(self.tableau.γ, dtype=sol_dtype)
+        α = jnp.array(self.tableau.α, dtype=sol_dtype)
+
+        def embed_lower(x):
+            out = np.zeros(
+                (self.tableau.num_stages, self.tableau.num_stages), dtype=x[0].dtype
+            )
+            for i, val in enumerate(x):
+                out[i + 1, : i + 1] = val
+            return jnp.array(out, dtype=sol_dtype)
+
+        a_lower = embed_lower(self.tableau.a_lower)
+        c_lower = embed_lower(self.tableau.c_lower)
+        m_sol = jnp.array(self.tableau.m_sol, dtype=sol_dtype)
+        m_error = jnp.array(self.tableau.m_error, dtype=sol_dtype)
+
+        # common L.H.S
+        eye_shape = jax.ShapeDtypeStruct(time_derivative.shape, dtype=sol_dtype)
+        A = (lx.IdentityLinearOperator(eye_shape) / (control * γ[0])) - (
+            lx.JacobianLinearOperator(
+                lambda y, args: terms.vf(t0, y, args), y0, args=args
+            )
+        )
+
+        u = jnp.zeros(
+            (self.tableau.num_stages,) + time_derivative.shape, dtype=sol_dtype
+        )
+
+        def use_saved_vf(u):
+            stage_0_vf = solver_state
+            stage_0_b = stage_0_vf + ((control * γ[0]) * time_derivative)
+            stage_0_u = lx.linear_solve(A, stage_0_b).value
+
+            u = u.at[0].set(stage_0_u)
+            start_stage = 1
+            return u, start_stage
+
+        if made_jump is False:
+            u, start_stage = use_saved_vf(u)
+        else:
+            u, start_stage = lax.cond(
+                eqxi.unvmap_any(made_jump), lambda u: (u, 0), use_saved_vf, u
+            )
+
+        def body(u, stage):
+            # Σ_j a_{stage j} · u_j
+            y0_increment = jnp.tensordot(a_lower[stage], u, axes=[[0], [0]])
+            vf = terms.vf(
+                t0 + (α[stage] * control),
+                y0 + y0_increment,
+                args,
+            )
+
+            # Σ_j (c_{stage j}/control) · u_j
+            c_scaled_control = jax.vmap(lambda c: c / control)(c_lower[stage])
+            vf_increment = jnp.tensordot(c_scaled_control, u, axes=[[0], [0]])
+
+            b = vf + vf_increment + ((control * γ[stage]) * time_derivative)
+            # solving Ax=b
+            stage_u = lx.linear_solve(A, b).value
```

Comment:
Can you add a `self.linear_solver` argument that specifies the linear solver to use? (C.f. where this appears elsewhere in Diffrax, e.g. as an attribute of `VeryChord`)